### PR TITLE
fix python3 (str/bytes) compatibility errors in untethered mode

### PR DESCRIPTION
**Pull request by 3ll3d00d**
some additional fixes required to get untethered mode to complete without exception

### CODE REVIEW

**Comment by 3ll3d00d** [comment]
[93de077](https://github.com/eoyilmaz/displaycal-py3/pull/262/commits/93de07787b40b93a7584e2b736417b46e1824314) -  enables untethered mode to actually complete & create  output

[48fcce5](https://github.com/eoyilmaz/displaycal-py3/pull/262/commits/48fcce51338badadc103e76c9d7fd9540298142b) - allows madvr 3dlut creation to complete

[ec62698](https://github.com/eoyilmaz/displaycal-py3/pull/262/commits/ec626983d4e75d18bd53efa88351199a5c5ce109) - fixes the verification report issue raised as #224

**Comment by 3ll3d00d** [comment]
https://github.com/eoyilmaz/displaycal-py3/pull/262/commits/4b1c5ee5dfe85a5facda84fabb6948a8cb19796d - fixes #267

File: DisplayCAL/display_cal.py [hunk 1]
```diff
@@ -9502,16 +9502,9 @@ def measurement_report_consumer(
...
-                and "XYZ_Y" in list(data.DATA_FORMAT.values())
-                and "XYZ_Z" in list(data.DATA_FORMAT.values())
-            ):
-                if (
-                    "LAB_L" not in list(data.DATA_FORMAT.values())
-                    and "LAB_A" not in list(data.DATA_FORMAT.values())
-                    and "LAB_B" not in list(data.DATA_FORMAT.values())
-                ):
+            data_formats = list(data.DATA_FORMAT.values())
+            if b"XYZ_X" in data_formats and b"XYZ_Y" in data_formats and b"XYZ_Z" in data_formats:
```
Comment (eoyilmaz) [inline DisplayCAL/display_cal.py:9506]: This is good but please keep the format, or better run "black" on this file

File: DisplayCAL/madvr.py [hunk 2]
```diff
@@ -1487,26 +1487,34 @@ def _wait_for_client(self, addr=None, timeout=1):
...
-                    if (
-                        client
-                        and client["component"] == "madTPG"
-                        and client.get("confirmed")
-                        and conn
-                        and self._send(conn, "StartTestPattern")
-                    ):
-                        self._client_socket = conn
-                        return True
+                    if client:
```
Comment (eoyilmaz) [inline DisplayCAL/madvr.py:1490]: please avoid burrying down this many if statements

**Review by eoyilmaz — APPROVED** [review]

File: DisplayCAL/madvr.py (hunk 2 above)
Comment (3ll3d00d) [reply DisplayCAL/madvr.py:1490]: the reason for this chain of conditionals is to get the precise logging for the reason, without this it just silently ignores responses so working out what broke is one for the debugger only (which is annoying given the thread model and constant flow of messages)

Comment (p5k369) [reply DisplayCAL/madvr.py:1490]: Hey @3ll3d00d its clear want you want to achive but you can get it much easier and without raising the complexity by inverting the statements. If something is not, log it.

File: DisplayCAL/display_cal.py (hunk 1 above)
Comment (3ll3d00d) [reply DisplayCAL/display_cal.py:9506]: black doesn't change the above, it does change assorted other parts of this file though so I can PR that if you like

File: DisplayCAL/madvr.py (hunk 2 above)
Comment (3ll3d00d) [reply DisplayCAL/madvr.py:1490]: happy to PR something but not entirely sure what the preference is, e.g. I could change to something like

```
                        has_sent_start = None
                        if (
                            client["component"] == b"madTPG"
                            and client.get("confirmed")
                            and conn
                        ):
                            has_sent_start = self._send(conn, "StartTestPattern")
                        safe_print(
                            f"client: {client['component']}, confirmed: {client.get('confirmed')}, connection: {conn}, sent_start: {has_sent_start}"
                        )
                        if has_sent_start:
                            return True
```

which still has all the info required in log but is somewhat simpler

Comment (p5k369) [reply DisplayCAL/madvr.py:1490]: I don't think you have to change it right now, its already merged. And its not about what you print but about your nested if statements, that is not considered best practice. "Flat is better than nested." https://peps.python.org/pep-0020/

you could for example also do something like:
```
....
if not client:
    safe_print(f'Ignoring client component : {component_}')
    return False
component_ = client['component']
if component_ != b'madTPG' and not client.get('confirmed'):
    safe_print('madTPG is unconfirmed')
    return False
....
```
Your function or method would have a straight path until it returns (in this case) True at the end. On its way it only handles the exceptions and exits early with the error message.
That way you will have one indent level and in general it is better readable for humans.

**Pull request merged**